### DISCUSSION TITLE
Crystal Tracker - A Game Boy Song Editor

### DISCUSSION THREAD

**Discussion by dannye** [fltk/fltk#1042] [Show and tell]
I've been working with FLTK for a little over two years to make a "MIDI style" Game Boy song editor, called [Crystal Tracker](https://github.com/dannye/crystal-tracker)!

It lets you compose songs in the style of Pokemon Gold/Silver/Crystal!

I have a lot of admiration for FLTK and great respect for the developers. In addition to FLTK, the editor synthesizes the song playback with the open source chiptune library called libopenmpt and drives the audio with PortAudio.

![Crystal Tracker - crystaltracked asm_012](https://github.com/user-attachments/assets/f82ce2bc-ad12-455a-b49d-22c836508a8b)
![Crystal Tracker - whackthediglett asm_013](https://github.com/user-attachments/assets/c5d2d761-87c4-448e-815b-8a540cd24c65)
![Crystal Tracker - credits asm_015](https://github.com/user-attachments/assets/f3f20428-f1e1-4729-8f2a-1956631c9457)
![Crystal Tracker - ecruteakcity asm_016](https://github.com/user-attachments/assets/e2d1795a-6bbe-4b74-ac8f-c714080e43d1)
![Crystal Tracker - club3 asm_017](https://github.com/user-attachments/assets/27228bf9-0879-4bf4-94ba-083f2d6fac3b)
![Crystal Tracker - dueltheme1 asm_018](https://github.com/user-attachments/assets/34c31dd2-bb07-473b-aecd-86e41b2f24b7)
![Crystal Tracker - viridiancity asm_019](https://github.com/user-attachments/assets/a9cb92d9-d5d1-4749-8a55-987edf2faf3c)
![Crystal Tracker - overworld asm_022](https://github.com/user-attachments/assets/5e8a49d8-27dc-450e-9027-923be0503726)

**Comment by Albrecht-S** [comment]
@dannye Thank you for your great work with FLTK and the posted screenshots. Do you allow us to pick one and add it to our official [application screenshots page](https://www.fltk.org/applications/shots.php)?

**Reply by dannye** [reply]
Absolutely! Thanks 😄

**Reply by dannye** [reply]
I updated all the screenshots with final touches from v0.8.0 (in case you already downloaded a copy).

**Reply by Albrecht-S** [reply]
Thanks, but where can we find these updated screenshots? Did you update the screenshots here, in this Discussion thread, or are they stored elsewhere? I did not yet find the time to pull the screenshots and to update our site, but it's still on my todo list.

**Reply by dannye** [reply]
I just edited the top post. There is also a copy of them uploaded to this [imgur album](https://imgur.com/a/u7JSBvS).

**Reply by Albrecht-S** [reply]
Thanks.

**Comment by MatthiasWM** [comment]
Looking really nice! Thankd for sharing.

**Comment by dannye** [comment]
I just released [v0.9.0](https://github.com/dannye/crystal-tracker/releases/tag/v0.9.0) of my song editor, using FLTK 1.4.5!

The big new features in this release are the Wave Editor and Drumkit Editor (instrument editors).

Here are a few screenshots:

<img width="610" height="394" alt="image" src="https://github.com/user-attachments/assets/2d61e84a-f8fa-4153-b272-302d6771cb84" />

<img width="610" height="394" alt="image" src="https://github.com/user-attachments/assets/3634ba22-ae49-4348-9eec-261c44ec88f4" />

<img width="610" height="394" alt="image" src="https://github.com/user-attachments/assets/9afd327f-eebd-497a-bc9d-193c7a4d2273" />

I want to give a huge thank you as always to ALL the FLTK developers for making this possible. You forever have my gratitude.
